pypsa/statistics.py

```diff
@@ -210,19 +266,52 @@ def aggregate_timeseries(df, weights, agg="sum"):
     return df.agg(agg)
 
 
+def filter_active_assets(n, c, df: [pd.Series, pd.DataFrame]):
+    """
+    For static values iterate over periods and concat values.
+    """
+    if not isinstance(n.snapshots, pd.MultiIndex) or isinstance(df, pd.DataFrame):
+        return df
+    per_period = {}
+    for p in n.investment_periods:
+        per_period[p] = df[n.get_active_assets(c, p).loc[df.index]]
+    return pd.concat(per_period, axis=1)
+
+
+def filter_bus_carrier(n, c, port, bus_carrier, df):
+    """
+    Filter the DataFrame for components which are connected to a bus with
+    carrier `bus_carrier`.
+    """
+    if bus_carrier is None:
+        return df
+
+    ports = n.df(c).loc[df.index, f"bus{port}"]
+    port_carriers = ports.map(n.buses.carrier)
+    if isinstance(bus_carrier, str):
```

Comment:
I think here it would be nice to have somehing like I did before, that we fist check if the `string` is in n.buses.carrier.unique() and if not we look if the bus_carriers contain the `string`